link_shorterner/src/shortener_link_handler.erl

```diff
@@ -0,0 +1,37 @@
+-module(shortener_link_handler).
+
+-export([init/2]).
+
+init(Req, State) ->
+    Url = get_request_url(Req),
+    ReqMethod = cowboy_req:method(Req),
+    {Status, ShortUrl} = handle_request(ReqMethod, Url),
+    Body = get_body_response(Status, ShortUrl),
+    Header = get_headers(),
+    Resp = cowboy_req:reply(Status, Header, Body, Req),
+    {ok, Resp, State}.
+
+handle_request(<<"POST">>, Url) -> 
+    {CreationStatus, ShortUrl} = shortener_shortener:short(Url),
+    HttpStatus =
+        case CreationStatus of
+            old -> 200; %ok
+            new -> 201  %created
+        end,
+    {HttpStatus, ShortUrl};
+
+handle_request(<<"GET">>, Url) ->
+    case shortener_shortener:get(Url) of
+        not_found -> {404, ""};
+        ShortUrl  -> {302, ShortUrl}
```

Comment:
I don't think this is enough to produce a redirect. Have you tested it with the browser? You'd have to be setting a `Location` header for it to work

Reply:
I am pretty sure you are right. The location header is automatically added with the rest handler but no by default with the http handler. Always test the functionality as an end user would do it.